cloud/aws/alb/monitors-alb.tf

```diff
@@ -0,0 +1,203 @@
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "are too low < ${var.no_healthy_instances_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.no_healthy_instances_disabled_warning, var.no_healthy_instances_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.no_healthy_instances_notifications_warning, var.no_healthy_instances_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "latency" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ALB latency"
+
+	program_text = <<-EOF
+		from signalfx.detectors.aperiodic import aperiodic
+		signal = data('TargetResponseTime', filter=filter('namespace', 'AWS/ApplicationELB') and filter('stat', 'mean') and filter('TargetGroup', '*') and (not filter('AvailabilityZone', '*')) and ${module.filter-tags.filter_custom})${var.latency_aggregation_function}.${var.latency_transformation_function}(over='${var.latency_transformation_window}')
+		above_or_below_detector(signal, ${var.latency_threshold_critical}, 'above', lasting('${var.latency_aperiodic_duration}', ${var.latency_aperiodic_percentage})).publish('CRIT')
+		above_or_below_detector(signal, ${var.latency_threshold_warning}, 'above', lasting('${var.latency_aperiodic_duration}', ${var.latency_aperiodic_percentage})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high > ${var.latency_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.latency_disabled_critical, var.latency_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.latency_notifications_critical, var.latency_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.latency_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.latency_disabled_warning, var.latency_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.latency_notifications_warning, var.latency_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "httpcode_5xx" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ALB 5xx errors"
+
+	program_text = <<-EOF
```

Comment:
we need to preserve the crazy guard about disabling alert if too few trafic so we need a custom threshold with AND operator adding request count minimal value